src/components/ui/frame.tsx

```diff
@@ -0,0 +1,58 @@
+"use client";
+import * as React from "react"
+import Image from "next/image";
+import type { ImgHTMLAttributes } from "react";
+import {
+    FrameUI,
+    fallbackFrameContext,
+} from "@frames.js/render";
+import { signFrameAction, FarcasterSigner } from '@frames.js/render/farcaster'
+import { useFrame } from "@frames.js/render/use-frame";
+import {AccountObjectType} from "@/stores/useAccountStore";
+
+export function FrameImageNext(
+    props: ImgHTMLAttributes<HTMLImageElement> & { src: string }
+): React.JSX.Element {
+    return (
+        <Image
+            {...props}
+            alt={props.alt ?? ""}
+            sizes="100vw"
+            height={0}
+            width={0}
+        />
+    );
+}
+
+
+export default function Frame( { homeFrameUrl, account } : { homeFrameUrl: string, account?: AccountObjectType }) {
+    const farcasterSigner: FarcasterSigner | undefined = account ? {
+        fid: account?.id,
```

Comment:
the fid for these herocast accounts is stored in `account.platformAccountId`.
to get the right type you can use 

```suggestion
        fid: Number(account?.platformAccountId),
```